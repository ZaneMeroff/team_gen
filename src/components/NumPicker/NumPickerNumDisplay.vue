<template>
  <section class="npnd-perimeter-container">

    <p class="npnd-num">{{ number }}</p>

  </section>
</template>

<script>
  export default {
    name: "NumPickerNumDisplay",
    props: {
      number: { type: Number, default: 0 },
    },
  }
</script>

<style scoped>

  .npnd-perimeter-container {
    height: 100%;
    width: 50%;
    display: flex;
    justify-content: center;
    align-items: center;
  }

  .npnd-num {
    font-size: 3.5rem;
    font-weight: bold;
    font-family: courier;
  }
</style>
