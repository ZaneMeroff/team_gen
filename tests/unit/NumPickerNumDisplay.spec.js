import { shallowMount } from '@vue/test-utils'

import NumPickerNumDisplay from "@/components/NumPicker/NumPickerNumDisplay"

describe("NumPickerNumDisplay", () => {

  describe("snapshots", () => {

    it("should render with default props", () => {
      const component = shallowMount(NumPickerNumDisplay)

      expect(component.element).toMatchSnapshot()
    })

    it("should set number prop", () => {
      const propsData = { number: 5 }
      const component = shallowMount(NumPickerNumDisplay, { propsData })

      expect(component.element).toMatchSnapshot()
    })
  })

  describe("props", () => {

    describe("number", () => {

      it("should have a default value", () => {
        const component = shallowMount(NumPickerNumDisplay)

        expect(component.vm.number).toEqual(0)
      })

      it("should accept a number", () => {
        const propsData = { number: 5 }
        const component = shallowMount(NumPickerNumDisplay, { propsData })

        expect(component.vm.number).toEqual(propsData.number)
      })
    })
  })
})
