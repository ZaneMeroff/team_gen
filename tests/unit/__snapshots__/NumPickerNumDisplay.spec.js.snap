// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumPickerNumDisplay snapshots should render with default html 1`] = `
<div
  class="npnd-perimeter-container"
>
  <p
    class="picker-title"
  >
    teams
  </p>
   
  <p
    class="npnd-num"
  >
    0
  </p>
</div>
`;

exports[`NumPickerNumDisplay snapshots should set number prop 1`] = `
<div
  class="npnd-perimeter-container"
>
  <p
    class="picker-title"
  >
    teams
  </p>
   
  <p
    class="npnd-num"
  >
    5
  </p>
</div>
`;

exports[`NumPickerNumDisplay snapshots should set pickerType prop 1`] = `
<div
  class="npnd-perimeter-container"
>
  <p
    class="picker-title"
  >
    players
  </p>
   
  <p
    class="npnd-num"
  >
    0
  </p>
</div>
`;
